### DISCUSSION TITLE
Editing files inside node_modules folder doesn't reflect on the browser

### DISCUSSION THREAD

**Discussion by JSsantana** [vercel/next.js#39873] [Help]
I have two projects, one is my main project and the other is a npm package in active development. After install the lib from npm, no matter what I edit inside node modules folder, it just don't reflect the changes. I even see a compile success message at the terminal but nothing changes at the browser, even refreshing, cleaning npm cache or restarting the project. 

Is there any config that I can use to enable nextjs recognize and hot reload the changes made inside a node-module folder?

**Comment by icyJoseph** [comment]
Try deleting `.next`, and running things again. Does that help? 

It could also be that if your package is compiled as CJS, and ESM, you gotta edit both.

**Reply by JSsantana** [reply]
Deleting .next works, but is there's another approach that I don't need to delete?

**Reply by icyJoseph** [reply]
Ah, perhaps check out this discussion, I haven't tried the solution described there, but surely Dan's proposal would've been good.

**Reply by byteab** [reply]
@JSsantana check this
https://github.com/vercel/next.js/discussions/33929#discussioncomment-2710035

**Comment by SivaSathyamurthy** [comment]
After editing files inside node modules, you need to do a any changes in your parent componenet, which will compile again and show the changes

**Comment by refanbanzhang** [comment]
Check your `node_modules` folder. If you find the `.cache` folder, this may be the reason.

You can delete this folder and restart you project for verification.

You can also edit webpack config file, set the `cache` to `false`.

**Comment by KeitelDOG** [comment]
In Next JS 16, I was able to edit next js app dev server file with building with webpack instead of turbopack which is used by defautl. In `package.json` just use a script:
`"dev:webpack": "next dev --webpack"`.

I tried everything else with no luck, but once I used webpack, the changes reflected and I was able to debug and change the behaviors I wanted for my experiment.

**Comment by js713-lab** [comment]
### Why edits inside `node_modules` don’t show up

Next/Webpack/Turbopack treat `node_modules` as **prebundled dependency code**. Many packages are pre-compiled, cached, and sometimes package-exported away from the files you’re editing. So changing files under `node_modules/<pkg>` is unreliable (and gets wiped on the next install).

**Better workflow for an in-progress local package:**

1. **`pnpm link` / `npm link` / `yarn link`** the local package into the app, or
2. Use **`pnpm.overrides` / npm `overrides`** / Yarn `resolutions` to point the dependency at `file:../your-package`, or
3. Use **Turborepo/Nx workspace** / monorepo packages.

Then run the app’s package in watch mode (`tsup --watch`, `unbuild -w`, etc.) so rebuilds propagate.

Also delete `.next` once after linking so the bundler drops stale dependency caches:

```bash
rm -rf .next
```
